interpreter/src/main/java/org/qbicc/interpreter/impl/Frame.java

```diff
@@ -0,0 +1,1937 @@
+            } else {
+                throw unsupportedType();
+            }
+        }
+
+        @Override
+        public ExecutableElement visit(Frame frame, VirtualMethodElementHandle node) {
+            Object instance = frame.require(node.getInstance());
+            if (instance instanceof VmObjectImpl) {
+                VmObjectImpl object = (VmObjectImpl) instance;
+                MethodElement methodElement = node.getExecutable();
+                MethodElement result = object.getVmClass().getTypeDefinition().resolveMethodElementVirtual(methodElement.getName(), methodElement.getDescriptor());
+                if (result == null) {
+                    VmClassImpl nsme = VmImpl.require().noSuchMethodErrorClass;
+                    throw new Thrown((VmThrowable) nsme.newInstance());
+                }
+                return result;
+            } else {
+                throw unsupportedType();
+            }
+        }
+
+        @Override
+        public ExecutableElement visit(Frame param, ConstructorElementHandle node) {
+            return node.getExecutable();
+        }
+
+        @Override
+        public ExecutableElement visit(Frame frame, StaticMethodElementHandle node) {
+            return node.getExecutable();
+        }
+    };
+
+    static final ValueHandleVisitor<Frame, Memory> GET_MEMORY = new ValueHandleVisitor<Frame, Memory>() {
+        @Override
+        public Memory visitUnknown(Frame frame, ValueHandle node) {
+            throw invalidHandleTypeForOp();
+        }
+
+        @Override
+        public Memory visit(Frame frame, ElementOf node) {
+            return node.getValueHandle().accept(this, frame);
+        }
+
+        @Override
+        public Memory visit(Frame frame, GlobalVariable node) {
+            VmImpl vm = (VmImpl) Vm.current();
+            return vm.getGlobal(node.getVariableElement());
+        }
+
+        @Override
+        public Memory visit(Frame frame, InstanceFieldOf node) {
+            return node.getValueHandle().accept(this, frame);
+        }
+
+        @Override
+        public Memory visit(Frame frame, LocalVariable node) {
+            throw new IllegalArgumentException("Memory not supported for local variables");
+        }
+
+        @Override
+        public Memory visit(Frame frame, MemberOf node) {
+            return node.getValueHandle().accept(this, frame);
+        }
+
+        @Override
+        public Memory visit(Frame frame, StaticField node) {
+            FieldElement variableElement = node.getVariableElement();
+            DefinedTypeDefinition enclosingType = variableElement.getEnclosingType();
+            VmThreadImpl thread = (VmThreadImpl) Vm.requireCurrentThread();
+            ClassContext cc = enclosingType.getContext();
+            VmClassLoaderImpl classLoader = (VmClassLoaderImpl) cc.getClassLoader();
+            if (classLoader == null) {
+                classLoader = thread.getVM().getBootstrapClassLoader();
+            }
```

Comment:
We probably want this to be wrapped in an api like `getClassLoaderIncludingBoot()` (with a better name) so we don't have to sprinkle the null check throughout the VM code.

Reply:
Yeah I agree, this needs to be improved.